### ISSUE TITLE
5744 segmentation fault on M2 Max

### ISSUE THREAD

**Issue by valeriylo** [vladkens/macmon#1]
Hi there!
I really wanted to try your utility, but when I installed it using homebrew, I got a segmentation error. When I run the `macmon` command in the terminal, the utility runs for a few seconds without displaying any information and then exits with a segmentation error.
When running `macmon raw`, I get a bit more information about the error - `5744 segmentation fault macmon raw`, without showing any UI.
I'm using a Macbook 16' M2 Max, Ventura 13.6

https://github.com/vladkens/macmon/assets/45401765/ec7e1537-f0db-47b0-8dd6-1a1bb85d6d98

**Comment by vladkens** [comment]
Hi, @valeriylo. Can you please run `macmon debug`. It will print internal metrics and probably crash too, but I will understand what sensors have an error.

PS. `sudo` not required to work

**Comment by valeriylo** [comment]
```
Chip: Apple M2 Max | Model: Mac14,6 | OS: macOS 13.6 (22G120) | proc 12:8:4

--- AppleARMIODevice -----------------------------------------------------------
                 voltage-states0: (v) 1 0 0
                 voltage-states0: (f) 0 0 0
                 voltage-states1: (v) 612 684 775 837 903 953
                 voltage-states1: (f) 0 0 0 0 0 0
            voltage-states1-sram: (v) 590 625 678 706 778 831
            voltage-states1-sram: (f) 912 1284 1752 2004 2256 2424
                voltage-states11: (v) 712
                voltage-states11: (f) 0
                voltage-states12: (v) 0
                voltage-states12: (f) 0
                voltage-states13: (v) 600 637 671 706 731 765 793 834 868 912 943 987 1025 1059 1059 1059 1059 1140 1140
                voltage-states13: (f) 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
           voltage-states13-sram: (v) 790 790 790 806 834 859 896 915 940 990 1021 1075 1103 1156 1156 1156 1156 1196 1196
           voltage-states13-sram: (f) 702 948 1188 1452 1704 1968 2208 2400 2568 2724 2868 3000 3132 3264 3360 3408 3504 3528 3696
                voltage-states14: (v) 400 628 653 690 728 800 862 906 937
                voltage-states14: (f) 0 444 612 808 968 1110 1236 1338 1398
           voltage-states14-sram: (v) 790 790 790 790 790 800 862 906 937
           voltage-states14-sram: (f) 0 444 612 808 968 1110 1236 1338 1398
                voltage-states15: (v) 400 628 653 690 731 790 843 881 912
                voltage-states15: (f) 0 444 612 808 968 1110 1236 1338 1398
           voltage-states15-sram: (v) 790 790 790 790 790 790 843 881 912
           voltage-states15-sram: (f) 0 444 612 808 968 1110 1236 1338 1398
                voltage-states16: (v) 400 625 643 684 721 781 834 871 915
                voltage-states16: (f) 0 444 612 808 968 1110 1236 1338 1398
           voltage-states16-sram: (v) 790 790 790 790 790 790 834 871 915
           voltage-states16-sram: (f) 0 444 612 808 968 1110 1236 1338 1398
                 voltage-states2: (v) 1 0 0 0
                 voltage-states2: (f) 0 0 0 0
                voltage-states31: (v) 690 750 828 878 906 656
                voltage-states31: (f) 552 760 980 1098 1200 400
                 voltage-states5: (v) 600 625 646 687 718 759 793 834 868 912 943 987 1018 1046 1046 1046 1046 1140 1140
                 voltage-states5: (f) 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
            voltage-states5-sram: (v) 790 790 790 806 834 859 896 915 940 990 1021 1075 1103 1156 1156 1156 1156 1196 1196
            voltage-states5-sram: (f) 702 948 1188 1452 1704 1968 2208 2400 2568 2724 2868 3000 3132 3264 3360 3408 3504 3528 3696
                 voltage-states8: (v) 593 628 687 737 806 862 915
                 voltage-states8: (f) 600 852 1104 1356 1596 1848 2100
                 voltage-states9: (v) 400 625 650 687 725 790 843 887 918
                 voltage-states9: (f) 0 444 612 808 968 1110 1236 1338 1398
            voltage-states9-sram: (v) 790 790 790 790 790 790 843 887 918
            voltage-states9-sram: (f) 0 444 612 808 968 1110 1236 1338 1398

--- IOReport -------------------------------------------------------------------
Energy Model ::  :: EACC_CPU0 (mJ) = 0.07W
Energy Model ::  :: EACC_CPU1 (mJ) = 0.08W
Energy Model ::  :: EACC_CPU2 (mJ) = 0.07W
Energy Model ::  :: EACC_CPU3 (mJ) = 0.05W
Energy Model ::  :: PACC0_CPU0 (mJ) = 0.00W
Energy Model ::  :: PACC0_CPU1 (mJ) = 0.00W
Energy Model ::  :: PACC0_CPU2 (mJ) = 0.00W
Energy Model ::  :: PACC0_CPU3 (mJ) = 0.00W
Energy Model ::  :: PACC1_CPU0 (mJ) = 0.04W
Energy Model ::  :: PACC1_CPU1 (mJ) = 0.01W
Energy Model ::  :: PACC1_CPU2 (mJ) = 0.00W
Energy Model ::  :: PACC1_CPU3 (mJ) = 0.00W
Energy Model ::  :: EACC_CPM (mJ) = 0.06W
Energy Model ::  :: PACC0_CPM (mJ) = 0.00W
Energy Model ::  :: PACC1_CPM (mJ) = 0.09W
Energy Model ::  :: EACC_CPU0_SRAM (mJ) = 0.01W
Energy Model ::  :: EACC_CPU1_SRAM (mJ) = 0.01W
Energy Model ::  :: EACC_CPU2_SRAM (mJ) = 0.01W
Energy Model ::  :: EACC_CPU3_SRAM (mJ) = 0.01W
Energy Model ::  :: PACC0_CPU0_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC0_CPU1_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC0_CPU2_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC0_CPU3_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC1_CPU0_SRAM (mJ) = 0.01W
Energy Model ::  :: PACC1_CPU1_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC1_CPU2_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC1_CPU3_SRAM (mJ) = 0.00W
Energy Model ::  :: EACC_CPM_SRAM (mJ) = 0.05W
Energy Model ::  :: PACC0_CPM_SRAM (mJ) = 0.00W
Energy Model ::  :: PACC1_CPM_SRAM (mJ) = 0.00W
Energy Model ::  :: EACC_CPU (mJ) = 0.42W
Energy Model ::  :: PACC0_CPU (mJ) = 0.01W
Energy Model ::  :: PACC1_CPU (mJ) = 0.17W
Energy Model ::  :: CPU Energy (mJ) = 0.58W
Energy Model ::  :: ECPUDTL00 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL01 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL02 (mJ) = 0.03W
Energy Model ::  :: ECPUDTL03 (mJ) = 0.02W
Energy Model ::  :: ECPUDTL04 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL05 (mJ) = 0.02W
Energy Model ::  :: ECPUDTL10 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL11 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL12 (mJ) = 0.03W
Energy Model ::  :: ECPUDTL13 (mJ) = 0.02W
Energy Model ::  :: ECPUDTL14 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL15 (mJ) = 0.03W
Energy Model ::  :: ECPUDTL20 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL21 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL22 (mJ) = 0.01W
Energy Model ::  :: ECPUDTL23 (mJ) = 0.02W
Energy Model ::  :: ECPUDTL24 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL25 (mJ) = 0.03W
Energy Model ::  :: ECPUDTL30 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL31 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL32 (mJ) = 0.01W
Energy Model ::  :: ECPUDTL33 (mJ) = 0.02W
Energy Model ::  :: ECPUDTL34 (mJ) = 0.00W
Energy Model ::  :: ECPUDTL35 (mJ) = 0.02W
Energy Model ::  :: PCPUDTL00 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL01 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL02 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL03 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL04 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL05 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL06 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL07 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL08 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL09 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL0a (mJ) = 0.00W
Energy Model ::  :: PCPUDTL0b (mJ) = 0.00W
Energy Model ::  :: PCPUDTL0c (mJ) = 0.00W
Energy Model ::  :: PCPUDTL0d (mJ) = 0.00W
Energy Model ::  :: PCPUDTL0e (mJ) = 0.00W
Energy Model ::  :: PCPUDTL0f (mJ) = 0.00W
Energy Model ::  :: PCPUDTL010 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL011 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL012 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL10 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL11 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL12 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL13 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL14 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL15 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL16 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL17 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL18 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL19 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL1a (mJ) = 0.00W
Energy Model ::  :: PCPUDTL1b (mJ) = 0.00W
Energy Model ::  :: PCPUDTL1c (mJ) = 0.00W
Energy Model ::  :: PCPUDTL1d (mJ) = 0.00W
Energy Model ::  :: PCPUDTL1e (mJ) = 0.00W
Energy Model ::  :: PCPUDTL1f (mJ) = 0.00W
Energy Model ::  :: PCPUDTL110 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL111 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL112 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL20 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL21 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL22 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL23 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL24 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL25 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL26 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL27 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL28 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL29 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL2a (mJ) = 0.00W
Energy Model ::  :: PCPUDTL2b (mJ) = 0.00W
Energy Model ::  :: PCPUDTL2c (mJ) = 0.00W
Energy Model ::  :: PCPUDTL2d (mJ) = 0.00W
Energy Model ::  :: PCPUDTL2e (mJ) = 0.00W
Energy Model ::  :: PCPUDTL2f (mJ) = 0.00W
Energy Model ::  :: PCPUDTL210 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL211 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL212 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL30 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL31 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL32 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL33 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL34 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL35 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL36 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL37 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL38 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL39 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL3a (mJ) = 0.00W
Energy Model ::  :: PCPUDTL3b (mJ) = 0.00W
Energy Model ::  :: PCPUDTL3c (mJ) = 0.00W
Energy Model ::  :: PCPUDTL3d (mJ) = 0.00W
Energy Model ::  :: PCPUDTL3e (mJ) = 0.00W
Energy Model ::  :: PCPUDTL3f (mJ) = 0.00W
Energy Model ::  :: PCPUDTL310 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL311 (mJ) = 0.00W
Energy Model ::  :: PCPUDTL312 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL00 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL01 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL02 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL03 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL04 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL05 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL06 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL07 (mJ) = 0.01W
Energy Model ::  :: PCPU1DTL08 (mJ) = 0.01W
Energy Model ::  :: PCPU1DTL09 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL0a (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL0b (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL0c (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL0d (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL0e (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL0f (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL010 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL011 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL012 (mJ) = 0.03W
Energy Model ::  :: PCPU1DTL10 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL11 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL12 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL13 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL14 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL15 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL16 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL17 (mJ) = 0.01W
Energy Model ::  :: PCPU1DTL18 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL19 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL1a (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL1b (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL1c (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL1d (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL1e (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL1f (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL110 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL111 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL112 (mJ) = 0.01W
Energy Model ::  :: PCPU1DTL20 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL21 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL22 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL23 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL24 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL25 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL26 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL27 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL28 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL29 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL2a (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL2b (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL2c (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL2d (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL2e (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL2f (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL210 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL211 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL212 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL30 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL31 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL32 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL33 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL34 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL35 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL36 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL37 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL38 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL39 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL3a (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL3b (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL3c (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL3d (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL3e (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL3f (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL310 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL311 (mJ) = 0.00W
Energy Model ::  :: PCPU1DTL312 (mJ) = 0.00W
Energy Model ::  :: GPU0 (mJ) = 0.06W
Energy Model ::  :: GPU CS0 (mJ) = 0.01W
Energy Model ::  :: GPU SRAM0 (mJ) = 0.00W
Energy Model ::  :: GPU CS SRAM0 (mJ) = 0.00W
Energy Model ::  :: ANE0 (mJ) = 0.00W
Energy Model ::  :: ISP0 (mJ) = 0.00W
Energy Model ::  :: AVE0 (mJ) = 0.00W
Energy Model ::  :: MSR0 (mJ) = 0.01W
Energy Model ::  :: DCS0 (mJ) = 1.03W
Energy Model ::  :: DRAM0 (mJ) = 0.51W
Energy Model ::  :: AMCC0 (mJ) = 0.79W
Energy Model ::  :: PCIe Port 0 Energy (uJ) = 0.00W
Energy Model ::  :: PCIe Port 1 Energy (uJ) = 0.00W
Energy Model ::  :: apciec2 Energy (uJ) = 0.00W
Energy Model ::  :: apciec0 Energy (uJ) = 0.00W
Energy Model ::  :: apciec1 Energy (uJ) = 0.00W
Energy Model ::  :: GPU Energy (nJ) = 0.08W
CPU Stats :: CPU Complex Performance States :: ECPU (24Mticks) =
CPU Stats :: CPU Complex Performance States :: ECPM (24Mticks) =
CPU Stats :: CPU Complex Performance States :: PCPU (24Mticks) =
CPU Stats :: CPU Complex Performance States :: PCPM (24Mticks) =
CPU Stats :: CPU Complex Performance States :: PCPU1 (24Mticks) =
CPU Stats :: CPU Complex Performance States :: PCPM1 (24Mticks) =
CPU Stats :: CPU Core Performance States :: ECPU00 (24Mticks) =
CPU Stats :: CPU Core Performance States :: ECPU01 (24Mticks) =
CPU Stats :: CPU Core Performance States :: ECPU02 (24Mticks) =
CPU Stats :: CPU Core Performance States :: ECPU03 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU00 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU01 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU02 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU03 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU10 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU11 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU12 (24Mticks) =
CPU Stats :: CPU Core Performance States :: PCPU13 (24Mticks) =
GPU Stats :: GPU Performance States :: GPUPH (24Mticks) =

--- IOHID ----------------------------------------------------------------------
[1]    9440 segmentation fault  macmon debug
```

**Comment by vladkens** [comment]
@valeriylo thank you for logs.

Should be fixed in new version (`v0.2.2`):

```sh
brew update && brew upgrade macmon
```

Most likely your Mac will not to show temperature values. As I understand SMC sensors available from macOS Sonoma, IOHID sensors which I use as fallback on Ventura works only for original M1 processors.

**Comment by valeriylo** [comment]
Works flawlessly now on `v0.2.2`! Thanks for your fix!
